compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -571,13 +572,13 @@ private static void createRecordType(MethodVisitor mv, BRecordType recordType, B
      * @param mv     method visitor
      * @param fields record fields to be added
      */
-    private static void addRecordFields(MethodVisitor mv, @Nilable List<BField> fields) {
+    private static void addRecordFields(MethodVisitor mv, @Nilable Map<String, BField> fields) {
         // Create the fields map
         mv.visitTypeInsn(NEW, LINKED_HASH_MAP);
         mv.visitInsn(DUP);
         mv.visitMethodInsn(INVOKESPECIAL, LINKED_HASH_MAP, "<init>", "()V", false);
 
-        for (BField optionalField : fields) {
+        for (BField optionalField : fields.values()) {
```

Comment:
Wondering if we should iterate through the entries and you the name at L586. Applicable in several places here and in JvmValueGen.java.

Reply:
Used `values()` since the value contains all the info we need + if iterating through the entry set, will have to refer to the field as `.value()` each time